src/huggingface_hub/file_download.py

```diff
@@ -844,11 +844,19 @@ def _create_relative_symlink(src: str, dst: str, new_blob: bool = False) -> None
     except OSError:
         pass
 
-    cache_dir = os.path.dirname(os.path.commonpath([src, dst]))
-    if are_symlinks_supported(cache_dir=cache_dir):
-        relative_src = os.path.relpath(src, start=os.path.dirname(dst))
+    try:
+        _support_symlinks = are_symlinks_supported(
+            os.path.dirname(os.path.commonpath([os.path.realpath(src), os.path.realpath(dst)]))
+        )
+    except PermissionError:
+        # Permission error means src and dst are not in the same folder (e.g. destination path has been provided
```

Comment:
```suggestion
        # Permission error means src and dst are not in the same volume (e.g. destination path has been provided
```

(guessing that's what we mean here)